indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -0,0 +1,150 @@
+import json
+
+from uuid import uuid4
+
+from flask import session, request, redirect, jsonify
+from sqlalchemy.orm import joinedload
+
+from indico.core.db import db
+from indico.modules.events.registration.controllers.management import RHManageRegFormBase, RHManageRegistrationBase
+from indico.modules.events.registration.models.items import RegistrationFormItemType, RegistrationFormItem
+from indico.modules.events.registration.models.registrations import Registration, RegistrationData
+from indico.modules.events.registration.models.form_fields import RegistrationFormFieldData
+from indico.modules.events.registration.views import WPManageRegistration
+from indico.web.flask.templating import get_template_module
+from indico.web.flask.util import url_for
+from indico.web.util import jsonify_data
+from MaKaC.common.cache import GenericCache
+
+
+cache = GenericCache('reglist-config')
+
+
+def _get_filters_from_request():
+    filters = {}
+    for key in request.form.iterkeys():
+        field_name = key.split('_')
+        if field_name[0] == 'radiogroup':
+            filters[int(field_name[1])] = request.form.getlist(key)
+    return filters
+
+
+def _filter_registration(query, filters):
+    if not filters:
+        return query
+
+    criteria = [db.and_(RegistrationFormFieldData.field_id == field_id,
+                        RegistrationData.data.op('#>>')('{}').in_(data_list))
+                for field_id, data_list in filters.iteritems()]
+    subquery = (RegistrationData.query
+                .with_entities(db.func.count(RegistrationData.registration_id))
+                .join(RegistrationData.field_data)
+                .filter(RegistrationData.registration_id == Registration.id)
+                .filter(db.or_(*criteria))
+                .correlate(Registration)
+                .as_scalar())
+    return query.filter(subquery == len(filters))
+
+
+def _query_registrations(regform, filters):
+    query = (Registration.query
+                         .with_parent(regform)
+                         .options(joinedload('data').joinedload('field_data').joinedload('field')))
+    return _filter_registration(query, filters)
+
+
+class RHRegistrationsListManage(RHManageRegFormBase):
+    """List all registrations of a specific registration form of an event"""
+
+    def _process(self):
+        session_key = 'reg_list_config_{}'.format(self.regform.id)
+        report_config_uuid = request.args.get('config')
+        if report_config_uuid:
+            configuration = cache.get(report_config_uuid)
+            if session_key in session:
+                session[session_key] = configuration
+            else:
+                session.setdefault(session_key, configuration)
+            return redirect(url_for('.manage_reglist', self.regform))
+        reg_list_config = session.get(session_key, {'items': [], 'filters': {}})
+        regform_items = RegistrationFormItem.find_all(RegistrationFormItem.id.in_(reg_list_config['items']))
```

Comment:
No matter if a field is deleted or not, I believe it should be available in the column selector, since we might have values for it in the registration data.